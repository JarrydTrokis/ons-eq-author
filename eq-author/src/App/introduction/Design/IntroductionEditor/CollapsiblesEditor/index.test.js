import React from "react";
import { shallow } from "enzyme";

import { CollapsiblesEditor } from "./";

describe("CollapsiblesEditor", () => {
  let props;
  beforeEach(() => {
    props = {
      collapsibles: [
        {
          id: "1",
          title: "title",
          description: "description",
        },
      ],
      createCollapsible: jest.fn(),
      moveCollapsible: jest.fn(),
      introductionId: "introId",
    };
  });

  it("should render", () => {
    const wrapper = shallow(<CollapsiblesEditor {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the collapsible editor and additional props for the collapsible given", () => {
    const wrapper = shallow(<CollapsiblesEditor {...props} />)
      .find("[data-test='collapsibles-list']")
      .renderProp("children")({ prop1: "1", prop2: 2 }, props.collapsibles[0]);

    expect(wrapper.props()).toMatchObject({
      collapsible: props.collapsibles[0],
      prop1: "1",
      prop2: 2,
    });
  });

  it("should create the collapsible when the add button is clicked", () => {
    shallow(<CollapsiblesEditor {...props} />)
      .find("[data-test='add-collapsible-btn']")
      .simulate("click");

    expect(props.createCollapsible).toHaveBeenCalledWith({
      introductionId: "introId",
    });
  });

  it("should move the collapsible when move is triggered", () => {
    shallow(<CollapsiblesEditor {...props} />)
      .find("[data-test='collapsibles-list']")
      .simulate("move", { id: "1", position: 2 });

    expect(props.moveCollapsible).toHaveBeenCalledWith({
      id: "1",
      position: 2,
    });
  });
});
