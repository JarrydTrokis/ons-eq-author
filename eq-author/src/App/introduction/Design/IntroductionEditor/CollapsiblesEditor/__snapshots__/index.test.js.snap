// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsiblesEditor should render 1`] = `
<div>
  <Reorder
    Transition={[Function]}
    data-test="collapsibles-list"
    list={
      Array [
        Object {
          "description": "description",
          "id": "1",
          "title": "title",
        },
      ]
    }
    onMove={[MockFunction]}
  >
    <Component />
  </Reorder>
  <CollapsiblesEditor__AddButton
    data-test="add-collapsible-btn"
    onClick={[Function]}
    variant="secondary"
  >
    Add collapsible
  </CollapsiblesEditor__AddButton>
</div>
`;
